### PR TITLE
feat: add retry support for copilot

### PR DESCRIPTION
**Pull request by darkskygit**
fix AFF-1066

### CODE REVIEW

**Comment by darkskygit** [comment]
* **#6947** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6947?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @darkskygit and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/server/src/plugins/copilot/controller.ts [hunk 1]
```diff
@@ -129,7 +136,6 @@ export class CopilotController {
     @CurrentUser() user: CurrentUser,
     @Req() req: Request,
     @Param('sessionId') sessionId: string,
-    @Query('messageId') messageId: string,
```
Comment (pengx17) [inline packages/backend/server/src/plugins/copilot/controller.ts:132]: is this a breaking change? Maybe older client will not work any longer with it. Shall we make it compatible for old client at least for one major version?

Comment (darkskygit) [reply packages/backend/server/src/plugins/copilot/controller.ts:132]: message id will get in `@Query() params: Record<string, string | string[]>`, no behavier change

File: packages/backend/server/tests/copilot.e2e.ts [hunk 2]
```diff
@@ -228,6 +228,61 @@ test('should be able to chat with api', async t => {
...
+      app,
+      token,
+      id,
+      randomUUID(),
+      promptName
+    );
+    const messageId = await createCopilotMessage(app, token, sessionId);
+    // chat 2 times
+    await chatWithText(app, token, sessionId, messageId);
+    await chatWithText(app, token, sessionId, messageId);
```
Comment (pengx17) [inline packages/backend/server/tests/copilot.e2e.ts:249]: In the real world case (like in chatgpt), the user may want to **edit** the last message and then resend. I think the current impl does not cover this case?

Comment (darkskygit) [reply packages/backend/server/tests/copilot.e2e.ts:249]: currently pr only cover the case about retry action or latest chat to get different response
edit message will impl in branching timeline like the chatgpt

**Review by pengx17 — APPROVED** [review]

**Pull request merged**